versioned/spi/src/main/java/org/projectnessie/versioned/NamedReferencesParams.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned;
+
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+
+/**
+ * Parameters that control the values that shall be returned via for each {@link NamedReference} via
+ * functionality to retrieve the named references.
+ */
+@Value.Immutable
+public interface NamedReferencesParams {
+
+  NamedReferencesParams DEFAULT = builder().build();
+
+  static ImmutableNamedReferencesParams.Builder builder() {
+    return ImmutableNamedReferencesParams.builder();
+  }
+
+  /**
+   * Name of the default branch to use as a reference for the computation of {@link
+   * NamedReference#getAheadBehind()}. If this parameter is not {@code null}, the branch must exist.
+   *
+   * @return name of the default branch. Can be {@code null}, if not needed.
+   */
+  @Nullable
+  String getDefaultBranchName();
```

Comment:
can we not just assume it's whatever the default branch is in the config instead of having to pass this explicitly?

Reply:
`DatabaseAdapter` doesn't know the `VersionStore` configuration

Reply:
I suppose in UI / CLI applications this branch does not have to be the "default" branch. A user may want to compare two arbitrary branches. How about renaming to something like `getRelatedReference()`? I guess it could be a tag too.

Reply:
I'm fine with adding that to `DatabaseAdapter`. There's technically no difference for the implementation.

But I'd rather leave that functionality out of the REST API for now to not make it too complicated to use.